src/redux/Products/products-reducer.js

```diff
@@ -0,0 +1,51 @@
+import * as actionTypes from "../shopping-types";
+
+export const getProductsReducer = (state = { products: [] }, action) => {
+    switch (action.type) {
+      case actionTypes.GET_PRODUCTS_REQUEST:
+        return {
+          loading: true,
+          products: [],
+        };
+      case actionTypes.GET_PRODUCTS_SUCCESS:
+        return {
+          products: action.payload,
+          loading: false,
+        };
+      case actionTypes.GET_PRODUCTS_FAIL:
+        return {
+          loading: false,
+          error: action.payload,
+        };
+        
```

Comment:
It should always maintain keys of state, use
```suggestion
        return {
        ...state,
          loading: false,
          error: action.payload,
        };
        
```

Reply:
Fix this everywhere